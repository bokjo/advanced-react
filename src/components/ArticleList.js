import React     from 'react';
import PropTypes from 'prop-types';
import Article   from './Article';


const ArticleList = (props) => {

  return (
    <div>
      {Object.values(props.articles).map((article) =>
        <Article
          key={article.id}
          article={article}
          store={props.store}
        />
      )}
    </div>
  );
};

ArticleList.propTypes = {

};

export default ArticleList;
