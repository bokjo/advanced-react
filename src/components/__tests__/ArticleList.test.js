import React       from 'react';
import ArticleList from '../ArticleList';

// import renderer    from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: { id: 'a', title: 'title a', date: 'date', body: 'body' },
      b: { id: 'b', title: 'title b', date: 'date', body: 'body' },
    },
  };

  it('List component renders correctly...', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );

    console.log(wrapper.node.props.children);

    // expect(wrapper.node.props.children.length).toBe(2);
    expect(wrapper.find('Article').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });

});
