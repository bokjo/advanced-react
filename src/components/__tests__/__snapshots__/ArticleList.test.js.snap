// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList List component renders correctly... 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <ArticleContainer
        article={
            Object {
                "body": "body",
                "date": "date",
                "id": "a",
                "title": "title a",
              }
        }
    />
    <ArticleContainer
        article={
            Object {
                "body": "body",
                "date": "date",
                "id": "b",
                "title": "title b",
              }
        }
    />
</div>,
  "nodes": Array [
    <div>
      <ArticleContainer
            article={
                  Object {
                        "body": "body",
                        "date": "date",
                        "id": "a",
                        "title": "title a",
                      }
            }
      />
      <ArticleContainer
            article={
                  Object {
                        "body": "body",
                        "date": "date",
                        "id": "b",
                        "title": "title b",
                      }
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ArticleList
        articles={
                Object {
                        "a": Object {
                          "body": "body",
                          "date": "date",
                          "id": "a",
                          "title": "title a",
                        },
                        "b": Object {
                          "body": "body",
                          "date": "date",
                          "id": "b",
                          "title": "title b",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "articles": Object {
            "a": Object {
              "body": "body",
              "date": "date",
              "id": "a",
              "title": "title a",
            },
            "b": Object {
              "body": "body",
              "date": "date",
              "id": "b",
              "title": "title b",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <ArticleContainer
                    article={
                              Object {
                                        "body": "body",
                                        "date": "date",
                                        "id": "a",
                                        "title": "title a",
                                      }
                    }
          />
          <ArticleContainer
                    article={
                              Object {
                                        "body": "body",
                                        "date": "date",
                                        "id": "b",
                                        "title": "title b",
                                      }
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <ArticleContainer
                    article={
                              Object {
                                        "body": "body",
                                        "date": "date",
                                        "id": "a",
                                        "title": "title a",
                                      }
                    }
          />
          <ArticleContainer
                    article={
                              Object {
                                        "body": "body",
                                        "date": "date",
                                        "id": "b",
                                        "title": "title b",
                                      }
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArticleList
    articles={
        Object {
            "a": Object {
              "body": "body",
              "date": "date",
              "id": "a",
              "title": "title a",
            },
            "b": Object {
              "body": "body",
              "date": "date",
              "id": "b",
              "title": "title b",
            },
          }
    }
/>,
}
`;
